### PR TITLE
GH-1631: fix missing @SendTo for cglib proxies

### PR DESCRIPTION
**Pull request by radiment**
Use handler.getBeanType() instead of bean.getClass(). Due to bean could be cglib proxy bean.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@radiment Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=1632)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=1632) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@radiment Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-kafka&pullRequestId=1632)!

File: spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java [hunk 1]
```diff
@@ -207,8 +207,9 @@ private void setupReplyTo(InvocableHandlerMethod handler) {
 			replyTo = extractSendTo(method.toString(), ann);
 		}
 		if (ann == null) {
-			ann = AnnotationUtils.getAnnotation(this.bean.getClass(), SendTo.class);
-			replyTo = extractSendTo(this.getBean().getClass().getSimpleName(), ann);
+			Class<?> beanType = handler.getBeanType();
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java:210]: I think we can reach the same solution just using a `ClassUtils.getUserClass()`.
But it's OK what you provide so far.
Now we need a test-case to prove the issue and fix for it.

Thanks

**Review by artembilan — CHANGES_REQUESTED** [review]

**Comment by garyrussell** [comment]
Thanks; merged as 90b9f625 after fixing checkstyle (imports).

**Comment by garyrussell** [comment]
Cherry-picked to 2.5.x, 2.4.x, 2.3.x.

**Pull request closed without merging**